Add Stackdriver Backend

winfieldj opened the pull request:
This PR adds a Stackdriver metrics backend. Collected Buildkite metrics available for the specified gcp project as:

custom.googleapis.com/buildlkite/total/[Metric_Name]

Queue_Name is specified via labels on a metric event.

lox commented:
Sorry for the slow reply! Will get this reviewed soon!

lox commented on README.md:
```diff
@@ -72,6 +72,10 @@ The Prometheus backend supports the following arguments
 
+The Stackdriver backend supports the following arguments
+
+* `-stackdriver-gcp-projectID`: The Google Cloud Platform to report metrics for
```
Could we go with all lowercase? 

```suggestion
* `-stackdriver-gcp-projectid`: The Google Cloud Platform to report metrics for
```

lox commented:
This brings in a huge amount of dependencies 🤔 I wonder what the best way to review these is. Other than that, this looks ok to me. 

Is a basic test possible?

winfieldj commented:
@lox 
* Updated stackdriver project id command flag to all lower case
* Updated to distinguish queues via Labels when recording metrics in stackdriver
* Refactored backend to support testing
* Added testing of backend
* was able to slim down some of the dependency change

lox commented:
This is great @winfieldj!

lox commented on backend/stackdriver.go:
```diff
@@ -0,0 +1,138 @@
+}
+
+// NewStackdriverBackend returns a new StackDriverBackend for the specified project
+func NewStackdriverBackend(gcpProjectID string) (*StackDriverBackend, error) {
```
Should be `StackDriver`?

lox commented on backend/stackdriver.go:
```diff
@@ -0,0 +1,138 @@
+	if err != nil {
+		return nil, fmt.Errorf("[NewStackdriverBackend] could not create stackdriver client: %v", err)
+	}
+	countTypes := []string{
```
Can we externalize these? We recently added `WaitingJobsCount` if not.

lox commented:
Minor comments, but looking 👌🏻

winfieldj commented:
@lox

* Changed init of metric types to use incoming metric names vs static / constants - will pick up new count 
  types as they become available.
* Added additional error logging

lox commented:
Thanks @winfieldj!
